src/main/scala/is/hail/distributedmatrix/BlockMatrix.scala

```diff
@@ -579,23 +579,30 @@ class BlockMatrix(val blocks: RDD[((Int, Int), BDM[Double])],
 case class BlockMatrixTransposeRDDPartition(index: Int, prevPartition: Partition) extends Partition
 
 private class BlockMatrixTransposeRDD(dm: BlockMatrix)
-  extends RDD[((Int, Int), BDM[Double])](dm.blocks.sparkContext, Seq[Dependency[_]](new OneToOneDependency(dm.blocks))) {
+  extends RDD[((Int, Int), BDM[Double])](dm.blocks.sparkContext, Nil) {
+
+  private val newPartitioner = dm.partitioner.transpose
+  
+  def transposePi(pi: Int): Int = dm.partitioner.coordinatesBlock(
```

Comment:
I think acronyms should be all capitals if the first letter is capital.  So VDS not Vds.  pi stands for partition index, so I'd write transposePI not transposePi.